### PR TITLE
fix: use `endwith()` instead of `endWith()`

### PR DESCRIPTION
**Pull request by jef**
### Description

Fixes method call error created by this: https://github.com/beetbox/beets/pull/3449

Moved the helper static functions to the `SubsonicUpdate` class and made them private static.

Added tests for various of URL input and made sure that it can create a token and format URL correctly.

### CODE REVIEW

**Comment by jef** [comment]
Working on creating some tests, so we don't run into this problem again.

**Comment by sampsyo** [comment]
So far, so good. :smiley:

File: beetsplug/subsonicupdate.py [hunk 1]
```diff
@@ -90,10 +49,51 @@ def __init__(self):
...
+
+        :return: The generated salt and hashed token
+        """
+        password = config['subsonic']['pass'].as_str()
+
+        # Pick the random sequence and salt the password
+        r = string.ascii_letters + string.digits
+        salt = "".join([random.choice(r) for _ in range(6)])
+        salted_password = password + salt
+        token = hashlib.md5(salted_password.encode('utf-8')).hexdigest()
```
Comment (jef) [inline beetsplug/subsonicupdate.py:64]: Also changed this, but hard to tell in GitHub diff...

Used to be `hashlib.md5().update(salted_password.encode('utf-8')).hexdigest()`. Seems like a proper use of `hashlib`.

File: beetsplug/subsonicupdate.py [hunk 2]
```diff
@@ -90,10 +49,51 @@ def __init__(self):
...
+    def __format_url():
+        """Get the Subsonic URL to trigger a scan. Uses either the url
+        config option or the deprecated host, port, and context_path config
+        options together.
+
+        :return: Endpoint for updating Subsonic
+        """
+
+        url = config['subsonic']['url'].as_str()
+        if url and url.endswith('/'):
```
Comment (jef) [inline beetsplug/subsonicupdate.py:79]: This was the main problem.

**Comment by sampsyo** [comment]
Nice; looks prefect to me! Thanks!!

**Comment by jef** [comment]
Thanks as always!

**Pull request merged**